plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -135,14 +141,15 @@ def check_all_and_associate(wanted, conduit, config, download_deferred, catalog)
                 # package file does not exist and downloading is not deferred.
                 if not download_deferred and not file_exists:
                     continue
-            catalog.add(unit, wanted[unit.unit_key_as_named_tuple].download_path)
-            if rpm_parse.signature_enabled(config):
-                try:
-                    rpm_parse.filter_signature(unit, config)
-                except PulpCodedException as e:
-                    _LOGGER.debug(e)
-                    continue
-            repo_controller.associate_single_unit(conduit.repo, unit)
+            if unit.id not in all_associated_units:
+                catalog.add(unit, wanted[unit.unit_key_as_named_tuple].download_path)
```

Comment:
I don't think this change is a good idea.  What is the unit is associated but not downloaded?  If not downloaded, we'll still need the catalog entry even if the unit is already associated.